samples/custom_module/static/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+if(NOT IREE_HAL_DRIVER_LOCAL_SYNC)
+  return()
+endif()
+if(NOT "static_sample" IN_LIST IREE_EXTERNAL_TOOLING_MODULES)
+  return()
+endif()
+
+message(WARNING "IREE custom_module/static sample compiled into all tools")
```

Comment:
We don't want this included in release builds, right? I _think_ the right spot to set `IREE_BUILD_SAMPLES=OFF` for that is here: https://github.com/openxla/iree/blob/f2fc7c5bb95a5e59f540d201474e53563f109f96/compiler/setup.py#L222-L247

(could also land and then watch the release build logs / try the draft release before promoting to stable)

Reply:
This is opt-in - unless you specify all the cmake args as in the README it doesn't build, so this is mostly a reminder that "hey you're building with the sample on fyi"

Reply:
Ahhhh gotcha. SG.